fv/bpf_test.go

```diff
@@ -637,6 +637,157 @@ func describeBPFTests(opts ...bpfTestOpt) bool {
 					cc.CheckConnectivity()
 				})
 
+				It("should not be able to spoof workloads IP by another workload", func() {
+					if testOpts.protocol != "udp" {
+						return
+					}
+
+					By("allowing any traffic", func() {
+						pol.Spec.Ingress = []api.Rule{
+							{
+								Action: "Allow",
+								Source: api.EntityRule{
+									Nets: []string{
+										"0.0.0.0/0",
+									},
+								},
+							},
+						}
+						pol = updatePolicy(pol)
+
+						cc.ExpectSome(w[1][0], w[0][0])
+						cc.ExpectSome(w[1][1], w[0][0])
+						cc.CheckConnectivity()
+					})
+
+					/* XXX temporary will be removed and replaced by enfocing it
+					 * XXX from felix in BPF
+					 */
+					By("setting strict RPF check", func() {
+						felixes[1].Exec("sysctl", "-w", "net.ipv4.conf.all.rp_filter=1")
+						felixes[1].Exec("sysctl", "-w", "net.ipv4.conf.default.rp_filter=1")
+					})
+
+					By("testing that packet sent by another workload is dropped", func() {
+						tcpdump := w[0][0].AttachTCPDump()
+						tcpdump.SetLogEnabled(true)
+						matcher := fmt.Sprintf("IP %s\\.30444 > %s\\.30444: UDP", w[1][0].IP, w[0][0].IP)
```

Comment:
Best to use backtick quotes for regexps to avoid double-escaping.

Could use `regep.QuoteMeta()`

Reply:
Wouldn't QuoteMate also quote `.*` and similar if I changed the regexp? Will probably stick with manual escaping in some form

Reply:
Fine not to use QuoteMeta; I still think backtick quotes are better for regexps:
```
`IP %s\.30444 > %s\.30444: UDP`
```
if more readable than
```
"IP %s\\.30444 > %s\\.30444: UDP"
```
(and the difference only grows with the complexity of the regexp.)